File: backend/src/main/java/com/qme/qmeapp/organization/dto/OrganizationResponse.java
package com.qme.qmeapp.organization.dto;

import lombok.Builder;
import lombok.Data;

import java.time.OffsetDateTime;
import java.util.UUID;

@Data
@Builder
public class OrganizationResponse {
    private UUID id;
    private String name;
    private String queueCode;
    private String openingHours;
    private String closingHours;
    private Integer waitTimeMin;
    private Integer waitTimeMax;
    private String location;
    private String contactNumber;
    private String status;
    private OffsetDateTime createdAt;
    
    // Additional fields populated when fetching details
    private Long totalWaitingCustomers;
    private Long totalServedToday;
}

File: mobile/app/src/main/java/com/qme/mobile/data/model/response/OrganizationResponse.kt
package com.qme.mobile.data.model.response

data class OrganizationResponse(
    val id: String,
    val name: String,
    val queueCode: String,
    val status: String,
    val openingHours: String?,
    val closingHours: String?,
    val waitTimeMin: Int?,
    val waitTimeMax: Int?,
    val location: String?,
    val contactNumber: String?,
    val totalWaitingCustomers: Int?,
    val totalServedToday: Int?,
    val photo: String?          // base64-encoded image (data URI), nullable for backward-compat
)
